Source/inv.cpp

```diff
@@ -2045,6 +2045,20 @@ bool UseInvItem(int pnum, int cii)
 				break;
 			}
 		}
+
+		// If speedlist item is not inventory, use same item at the end of the speedlist if exists.
+		if (speedlist) {
+			for (int i = INVITEM_BELT_LAST - INVITEM_BELT_FIRST; i > cii - INVITEM_BELT_FIRST && *sgOptions.Gameplay.autoRefillBelt; i--) {

```

Comment:
The auto-refill condition doesn't really belong in the for loop conditions in either loop. In this case you've already got a wrapping if statement that is a better place for the check, it could even be moved further up and guard both loops.
```suggestion
		if (speedlist && *sgOptions.Gameplay.autoRefillBelt) {
			for (int i = INVITEM_BELT_LAST - INVITEM_BELT_FIRST; i > cii - INVITEM_BELT_FIRST; i--) {
```

Pretty sure you could also handle the case where someone has a belt full of health pots and uses hotkey 3. Currently this will not use the potions in slot 1 and 2, you could run through the entire belt range (in reverse order like you're currently doing) skipping the current index.
```suggestion
		if (speedlist && *sgOptions.Gameplay.autoRefillBelt) {
			for (int i = INVITEM_BELT_LAST - INVITEM_BELT_FIRST; i >= 0; i--) {
				if (i == c) { continue; }
```

Reply:
Thanks, I agree that `autoRefillBelt` check shouldn't be in loop.  I moved it but must not have pushed before the PR.

I didn't loop around to the beginning of the belt intentionally.  My thought process was that this feature would give illusion that belt items will slide down to auto fill the slot used.  I think if someone used a similar item further in the belt that it would be intentional and may not want the items to "slide" to the right.  What's your thought on this?

Reply:
eh, I'm not fussed either way. personally I'd prefer using the item I'm mashing last but I wouldn't go to the effort of trying to make belt items drain to the centre for visual effect. I'm sure others would prefer the implementation you currently have.